src/views/AnnualReport.vue

```diff
@@ -840,53 +871,49 @@ export default {
             annualGeneralMeetingDate: this.agmDate || null, // API doesn't validate empty string
             didNotHoldAgm: this.noAgm || false,
             annualReportDate: this.asOfDate,
+            // NB: there was an enrichment ticket to populate offices and directors here
             offices: {
-              registeredOffice: {
-                deliveryAddress: this.addresses.registeredOffice['deliveryAddress'],
-                mailingAddress: this.addresses.registeredOffice['mailingAddress']
-              }
+              registeredOffice: this.updatedAddresses.registeredOffice
             },
-            directors: this.allDirectors.filter(el => el.cessationDate === null)
+            directors: this.updatedDirectors.filter(el => el.cessationDate === null)
           }
         }
-      } else if (this.isBComp) {
+      }
+
+      if (this.isBComp) {
         annualReport = {
           annualReport: {
             annualReportDate: this.asOfDate,
             nextARDate: this.dateToUsableString(new Date(this.nextARDate)),
+            // NB: there was an enrichment ticket to populate offices and directors here
             offices: {
-              registeredOffice: {
-                deliveryAddress: this.registeredAddress['deliveryAddress'],
-                mailingAddress: this.registeredAddress['mailingAddress']
-              },
-              recordsOffice: {
-                deliveryAddress: this.recordsAddress['deliveryAddress'],
-                mailingAddress: this.recordsAddress['mailingAddress']
-              }
+              registeredOffice: this.originalAddresses.registeredOffice,
+              recordsOffice: this.originalAddresses.recordsOffice
             },
-            directors: this.directors
+            directors: this.originalDirectors
           }
         }
       }
 
-      if (this.hasFilingCode(FilingCodes.DIRECTOR_CHANGE_OT) ||
-        this.hasFilingCode(this.freeDirectorChangeFeeCodemap)) {
+      // non-BCOMP only
+      if (
+        this.hasFilingCode(FilingCodes.DIRECTOR_CHANGE_OT) ||
+        this.hasFilingCode(FilingCodes.FREE_DIRECTOR_CHANGE_OT)
+      ) {
         changeOfDirectors = {
           changeOfDirectors: {
-            directors: this.allDirectors
+            directors: this.updatedDirectors
           }
         }
       }
 
-      if (this.hasFilingCode(FilingCodes.ADDRESS_CHANGE_OT) && this.addresses) {
```

Comment:
"this.addresses" is always true (it could be an empty object)

Anyway, if the user has not made any changes then there is no filing code, so it was redundant (or maybe a safety check no longer needed).